### PR TITLE
Add phase tap to twt modification

### PR DESCRIPTION
**Pull request by Meklo**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/entities/equipment/modification/TwoWindingsTransformerModificationEntity.java [hunk 1]
```diff
@@ -216,8 +299,11 @@ public TwoWindingsTransformerModificationInfos toModificationInfos() {
     private TwoWindingsTransformerModificationInfos.TwoWindingsTransformerModificationInfosBuilder<?, ?> toTwoWindingsTransformerModificationInfosBuilder() {
 
         List<TapChangerStepCreationEmbeddable> ratioTapChangerStepsEmbeddable = null;
+        List<TapChangerStepCreationEmbeddable> phaseTapChangerStepsEmbeddable = null;
         if (getTapChangerSteps() != null && getTapChangerSteps().size() > 0) {
             ratioTapChangerStepsEmbeddable = getTapChangerSteps().stream().filter(step -> step.getTapChangerType().equals(TapChangerType.RATIO)).sorted(Comparator.comparing(TapChangerStepCreationEmbeddable::getIndex)).collect(Collectors.toList());
+            phaseTapChangerStepsEmbeddable = getTapChangerSteps().stream().filter(step -> step.getTapChangerType().equals(TapChangerType.PHASE)).sorted(Comparator.comparing(TapChangerStepCreationEmbeddable::getIndex)).collect(Collectors.toList());
```
Comment (archendel) [inline src/main/java/org/gridsuite/modification/server/entities/equipment/modification/TwoWindingsTransformerModificationEntity.java:305]: try to use groupingBy instead or browsing the list twice ?

File: src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java [hunk 2]
```diff
@@ -1003,7 +1003,7 @@ public void shouldGetPosition() {
         var network2 = networkStoreService.getNetwork(TEST_NETWORK_MIXED_TOPOLOGY_ID);
         var vl = network.getVoltageLevel("v2");
         var vl2 = network2.getVoltageLevel("v2");
-        assertEquals(9, vl.getConnectableCount());
+        assertEquals(10, vl.getConnectableCount());
```
Comment (archendel) [inline src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java:1006]: Why has it increased ?

File: src/test/java/org/gridsuite/modification/server/modifications/TwoWindingsTransformerModificationTest.java [hunk 3]
```diff
@@ -228,7 +286,10 @@ public void testCreateWithErrors() throws Exception {
     public void testRatioTapChangerModification() throws Exception {
         TwoWindingsTransformerModificationInfos twoWindingsTransformerModificationInfos = TwoWindingsTransformerModificationInfos.builder()
                 .equipmentId("trf1")
-                        .build();
+                .phaseTapChanger(PhaseTapChangerModificationInfos.builder()
+                        .enabled(new AttributeModification<Boolean>(false, OperationType.SET))
+                        .build())
```
Comment (archendel) [inline src/test/java/org/gridsuite/modification/server/modifications/TwoWindingsTransformerModificationTest.java:291]: Should it be needed ?

File: src/test/java/org/gridsuite/modification/server/modifications/TwoWindingsTransformerModificationTest.java [hunk 4]
```diff
@@ -337,5 +398,125 @@ public void testRatioTapChangerModification() throws Exception {
...
 
     }
+
+    @Test
+    public void testPhaseTapChangerModification() throws Exception {
+        TwoWindingsTransformerModificationInfos twoWindingsTransformerModificationInfos = TwoWindingsTransformerModificationInfos.builder()
+                .equipmentId("trf2")
+                .ratioTapChanger(RatioTapChangerModificationInfos.builder()
+                        .enabled(new AttributeModification<Boolean>(false, OperationType.SET))
+                        .build())
```
Comment (archendel) [inline src/test/java/org/gridsuite/modification/server/modifications/TwoWindingsTransformerModificationTest.java:408]: should it be needed ?

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: just minor changes requested
tests: see Front PR

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: just a question to adress
tests: ok

**Review by dbraquart — APPROVED** [review]
code: ok
tests: not done since last review

**Review by archendel — APPROVED** [review]

**Review by dbraquart — APPROVED** [review]
code: ok
tests: ok

**Pull request merged**